operator/webhook/addons/certmanager/main.py

```diff
@@ -0,0 +1,156 @@
+import logging
+from typing import Mapping, List
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _new_certificate(obj) -> Mapping:
+    metadata = obj["metadata"]
+
+    name = metadata["name"]
+
+    namespace = metadata["namespace"]
+
+    annotations = metadata.get("annotations", None)
+    if annotations is None:
+        _LOGGER.debug(
+            "IntegrationRoute does not contain metadata.annotations. No certificate will be generated."
+        )
+        return {}
+
+    cert_manager_io_annotations = [
+        annotation for annotation in annotations if "cert-manager.io" in annotation
+    ]
+
+    if cert_manager_io_annotations:
+        cluster_issuer = annotations.get("cert-manager.io/cluster-issuer", None)
+        if cluster_issuer is None:
+            _LOGGER.error(
+                "IntegrationRoute does not contain metadata.annotations.cert-manager.io/cluster-issuer"
+            )
+            return {}
+
+        common_name = annotations.get("cert-manager.io/common-name", None)
+        if common_name is None:
+            _LOGGER.error(
+                "IntegrationRoute does not contain metadata.annotations.cert-manager.io/common-name"
+            )
+            return {}
+    else:
+        return {}
+
+    keystore_type = _get_keystore_type(obj)
+    password_secret_ref_name = _get_password_secret_ref_name(obj)
+
+    alt_names = (
+        [n.strip() for n in annotations.get("cert-manager.io/alt-names").split(",")]
+        if annotations.get("cert-manager.io/alt-names", None) is not None
+        else []
+    )

```

Comment:
So for this annotation: `"cert-manager.io/alt-names": ","`

Seems this logic would end up adding two empty strings to the list of `dnsNames`.  Is that the expected behavior?

An alternative for filtering blank strings could be:
```
alt_names = [n.strip() for n in annotations.get("cert-manager.io/alt-names", []).split(",") if n.strip()]
```

If calling `strip()` twice bothers you, you could get fancier:
```
alt_names = [stripped_name for n in annotations.get("cert-manager.io/alt-names", []).split(",") if (stripped_name := n.strip())]
```

Reply:
As this string splitting logic is repeated across multiple annotations, might make sense to extract it to a common function, and possibly add some unit tests for some of the obvious edge cases (no need to go overboard).

Reply:
Fixed, moved to a common function, and added tests